### ISSUE TITLE
`unused-async` (`RUF029`) - false positive in `.pyi` files

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#11018]
```py
async def foo() -> int: ... # error: Function `foo` is declared `async`, but doesn't `await` or use `async` features.
```

this is a stub for a function that returns `Coroutine[Any, Any, int]`. removing the `async` will result in an incorrect return type

**Comment by charliermarsh** [comment]
I think we should omit stubs entirely.
